src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.ts

```diff
@@ -66,23 +66,51 @@ export class ContentTypeFieldsVariablesComponent implements OnInit, OnChanges, O
      * @param {FieldVariable} variable
      * @memberof ContentTypeFieldsVariablesComponent
      */
-    deleteVariable(fieldIndex: number): void {
-        const params: FieldVariableParams = {
-            contentTypeId: this.field.contentTypeId,
-            fieldId: this.field.fieldId,
-            variable: this.fieldVariables[fieldIndex]
-        };
-        this.fieldVariablesService.delete(params).pipe(take(1)).subscribe(
-            () => {
-                this.fieldVariables = this.fieldVariables.filter(
-                    (_item: FieldVariable, index: number) => index !== fieldIndex
-                );
-                this.canSaveFields.splice(fieldIndex, 1);
-            },
-            (err: ResponseView) => {
-                this.dotHttpErrorManagerService.handle(err).pipe(take(1)).subscribe();
-            }
-        );
+    deleteVariable(fieldVariable: FieldVariable, fieldIndex: number): void {
+        if (fieldVariable.key !== '') {
+            const params: FieldVariableParams = {
+                contentTypeId: this.field.contentTypeId,
+                fieldId: this.field.fieldId,
+                variable: this.fieldVariables[fieldIndex]
+            };
+            this.fieldVariablesService.delete(params).pipe(take(1)).subscribe(
+                () => {
+                    this.fieldVariables = this.fieldVariables.filter(
+                        (_item: FieldVariable, index: number) => index !== fieldIndex
+                    );
+                },
+                (err: ResponseView) => {
+                    this.dotHttpErrorManagerService.handle(err).pipe(take(1)).subscribe();
+                }
+            );
+        } else {
+            this.fieldVariables = this.fieldVariables.filter(
+                (_item: FieldVariable, index: number) => index !== fieldIndex
+            );
+        }
+    }
+
+    /**
+     * Handle Edit button event
+     * @param {Number} fieldIndex
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    editVariableAction(fieldIndex: number): void {
+        const key = document.getElementById(`content-type-fields__variables-key-${fieldIndex}`);
+        key.click();
```

Comment:
Why this?

Reply:
when the user click on the `edit` button, we select the `key` field, which is hidden by default by the `table` component